sklearn/cluster/tests/test_feature_agglomeration.py

```diff
@@ -16,8 +16,10 @@ def test_feature_agglomeration():
                                       pooling_func=np.mean)
     agglo_median = FeatureAgglomeration(n_clusters=n_clusters,
                                         pooling_func=np.median)
-    assert_no_warnings(agglo_mean.fit, X)
-    assert_no_warnings(agglo_median.fit, X)
+    with pytest.warns(None) as record:
+        agglo_mean.fit(X)
+        agglo_median.fit(X)
+    assert not len(record)
```

Comment:
```suggestion
    with pytest.warns(None) as record:
        agglo_mean.fit(X)
    assert not len(record)
    with pytest.warns(None) as record:
        agglo_median.fit(X)
    assert not len(record)
```

while more verbose, it makes it faster to identify which of the 2 calls generates a warning in case of failure.